src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -15,88 +15,117 @@
+    SPELL_SUMMON_VOID_SENTINEL_SUMMONER_VISUAL  = 45989,
+    SPELL_SUMMON_VOID_SENTINEL                  = 45988,
+    SPELL_TRANSFORM_VISUAL_MISSILE              = 46205,
+    TRANSFORM_VISUAL_MISSILE_1                  = 46208,
+    TRANSFORM_VISUAL_MISSILE_2                  = 46178,
+    SPELL_OPEN_PORTAL                           = 45977,
+    SPELL_OPEN_PORTAL_2                         = 45976,
 
-    // Shadowsword Fury Mage's spells
-    SPELL_FEL_FIREBALL           = 46101,
-    SPELL_SPELL_FURY             = 46102,
+    //Dark Fiend Spells
+    SPELL_DARKFIEND_DAMAGE                      = 45944,
+    SPELL_DARKFIEND_VISUAL                      = 45936,
+    SPELL_DARKFIEND_SKIN                        = 45934,
 
     // Void Sentinel's spells
-    SPELL_SHADOW_PULSE           = 46087,
-    SPELL_VOID_BLAST             = 46161,
+    SPELL_SHADOW_PULSE_PERIODIC                 = 46086,
+    SPELL_VOID_BLAST                            = 46161,
 
-    // Void Spawn's spells
-    SPELL_SHADOW_BOLT_VOLLEY     = 46082,
+    //Black Hole Spells
+    SPELL_BLACKHOLE_SPAWN                       = 46242,
+    SPELL_BLACKHOLE_GROW                        = 46228
+};
 
-    //Dark Fiend Spells
-    SPELL_DARKFIEND_AOE          = 45944,
-    SPELL_DARKFIEND_VISUAL       = 45936,
-    SPELL_DARKFIEND_SKIN         = 45934,
+enum Phases
+{
+    PHASE_ONE                                   = 1,
+    PHASE_TWO                                   = 2
+};
 
-    //Black Hole Spells
-    SPELL_BLACKHOLE_SPAWN        = 46242,
-    SPELL_BLACKHOLE_GROW         = 46228
+enum Misc
+{
+    MAX_VOID_SPAWNS                             = 6,
+    MAX_SUMMON_BLOOD_ELVES                      = 4,
+    MAX_SUMMON_DARK_FIEND                       = 8
 };
 
-enum Events
+uint32 const SummonDarkFiendSpells[MAX_SUMMON_DARK_FIEND] =
 {
-    // M'uru
-    EVENT_DARKNESS = 1,
-    EVENT_SUMMON_HUMANOIDS,
-    EVENT_SUMMON_SENTINEL,
-    EVENT_PHASE_TRANSITION, // Delayed phase transition.
-    EVENT_ENRAGE,
-
-    // Entropius
-    EVENT_SUMMON_BLACK_HOLE
+    SPELL_SUMMON_DARK_FIEND_0,
+    SPELL_SUMMON_DARK_FIEND_1,
+    SPELL_SUMMON_DARK_FIEND_2,
+    SPELL_SUMMON_DARK_FIEND_3,
+    SPELL_SUMMON_DARK_FIEND_4,
+    SPELL_SUMMON_DARK_FIEND_5,
+    SPELL_SUMMON_DARK_FIEND_6,
+    SPELL_SUMMON_DARK_FIEND_7
 };
 
-enum Phases
+uint32 const SummonBloodElvesSpells[MAX_SUMMON_BLOOD_ELVES] =
 {
-    PHASE_ONE = 1,
-    PHASE_TWO,
+    SPELL_SUMMON_BERSERKER,
+    SPELL_SUMMON_BERSERKER_2,
+    SPELL_SUMNON_FURY_MAGE,
+    SPELL_SUMNON_FURY_MAGE_2
 };
 
-enum CreatureGroups
+class VoidSpawnSummon : public BasicEvent
 {
-    CREATURE_GROUP_HUMANOIDS,
-    CREATURE_GROUP_DARKFIENDS
+    public:
+        explicit VoidSpawnSummon(Creature* owner)
+            : _owner(owner)
+        {
+        }
+
+        bool Execute(uint64 /*time*/, uint32 /*diff*/)
+        {
+            _owner->CastSpell((Unit*)NULL, SPELL_SUMMON_VOID_SENTINEL, true);
```

Comment:
Then "pretty much everywhere else" is terrible. `NULL` is type unsafe (it's an integer constant) and all should be avoided for pointer types. `nullptr` does the same thing but is actually pointer typed.